src/test/java/net/testudobank/tests/MvcControllerIntegTest.java

```diff
@@ -591,6 +596,201 @@ public void testReversalOfSimpleWithdraw() throws SQLException, ScriptException,
+   
+    //Verify that customer1's balance decreased by $100. 
+    assertEquals((CUSTOMER1_BALANCE_IN_PENNIES - TRANSFER_AMOUNT_IN_PENNIES), (int)customer1Data.get("Balance"));
+
+    //Verify that customer2's balance increased by $100.
+    assertEquals((CUSTOMER2_BALANCE_IN_PENNIES + TRANSFER_AMOUNT_IN_PENNIES), (int)customer2Data.get("Balance"));
+
+    //Check that transfer request goes through.
+    assertEquals("account_info", returnedPage);
+
+  }
+
+  /**
+   * This test is written to check the scenario where a transfer between the sender and the recipient occurs when
+   * the recipient is in overdraft. The sender will have $1000 in the bank account and will send $100. The recipient
+   * will have an overdraft balance of $101. Receiving $100 should make the sender have an updated overdraft balance of
+   * $1. 
+   * 
+   * The sender's balance should decrease by the transfer amount.
+   */
+
+  @Test
+  public void testTransferPaysOffOverdraftBalance() throws SQLException, ScriptException { 
+    
+   //Initialize customer1 with a balance of $1000. Balance will be represented as pennies in DB.
+   double CUSTOMER1_BALANCE = 1000;
+   int CUSTOMER1_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER1_ID, CUSTOMER1_PASSWORD, CUSTOMER1_FIRST_NAME, CUSTOMER1_LAST_NAME, CUSTOMER1_BALANCE_IN_PENNIES);
+
+   //Initialize customer2 with a balance of $100. Balance will be represented as pennies in DB. 
+   double CUSTOMER2_BALANCE = 100;
+   int CUSTOMER2_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER2_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER2_ID, CUSTOMER2_PASSWORD, CUSTOMER2_FIRST_NAME, CUSTOMER2_LAST_NAME, CUSTOMER2_BALANCE_IN_PENNIES);
+
+   //Amount to transfer
+   int TRANSFER_AMOUNT = 100;
+   int TRANSFER_AMOUNT_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(TRANSFER_AMOUNT);
+
+
+   //Initializing users for the transfer
+   User CUSTOMER1 = new User();
+   User CUSTOMER2 = new User();
+   CUSTOMER1.setUsername(CUSTOMER1_ID);
+   CUSTOMER1.setPassword(CUSTOMER1_PASSWORD);
+
+   double CUSTOMER2_AMOUNT_TO_WITHDRAW = 201;
+   int CUSTOMER2_AMOUNT_TO_WITHDRAW_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER2_AMOUNT_TO_WITHDRAW);
+   int CUSTOMER2_EXPECTED_OVERDRAFT_BALANCE = MvcControllerIntegTestHelpers.applyOverdraftInterest(CUSTOMER2_BALANCE_IN_PENNIES - CUSTOMER2_AMOUNT_TO_WITHDRAW_IN_PENNIES) * -1;
+   CUSTOMER2.setUsername(CUSTOMER2_ID);
+   CUSTOMER2.setPassword(CUSTOMER2_PASSWORD);
+   CUSTOMER2.setAmountToWithdraw(CUSTOMER2_AMOUNT_TO_WITHDRAW);
+
+   CUSTOMER1.setWhoToTransfer(CUSTOMER2_ID);
+   CUSTOMER1.setAmountToTransfer(TRANSFER_AMOUNT);
+
+    //Withdraw $201 from Customer2's account to cause an overdraft of $101.
+    controller.submitWithdraw(CUSTOMER2);
```

Comment:
No need to do these withdraws to cause customer2 to go into overdraft. You can make customer2 start in the overdraft state.

You can just use the other `addCustomerToDB()` helper method that takes more inputs to initialize customer2 with a main balance of $0 and overdraft balance of $101. Look at the `testDepositOverdraftClearedWithExcess()` integ test for an example.